### PR TITLE
[Feature] Support ingesting nested array from parquet

### LINKED ISSUE #16403: [Feature] Support ingesting nested array from parquet
## Feature request

**Is your feature request related to a problem? Please describe.**

broker load currently not support ingesting nested array from parquet
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16403

## Problem Summary(Required) ：

broker load currently not support ingesting nested array from parquet, this pr add support to it.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
- [x] 2.5 
- [x] 2.4
- [ ] 2.3
- [ ] 2.2

### CODE REVIEW

File: be/src/exec/parquet_scanner.cpp [hunk 1]
```diff
@@ -259,23 +272,17 @@ Status ParquetScanner::convert_array_to_column(ConvertFunc conv_func, size_t num
...
-        if (st.ok()) {
-            // in some scene such as string length exceeds limit, the column will be set NULL, so we need reset has_null
-            if (column->is_nullable()) {
-                down_cast<NullableColumn*>(column.get())->update_has_null();
-            }
+
+    auto* filter_data = (&_chunk_filter.front()) + _chunk_start_idx;
+    auto st = conv_func(array, _batch_start_idx, num_elements, data_column, _chunk_start_idx, null_data, filter_data,
+                        &_conv_ctx);
+    if (st.ok()) {
```
Comment (chaoyli) [inline be/src/exec/parquet_scanner.cpp:279]:
```suggestion
    if (st.ok() && column->is_nullable())
```

File: be/src/exec/parquet_scanner.cpp [hunk 2]
```diff
@@ -244,6 +244,19 @@ Status ParquetScanner::new_column(const arrow::DataType* arrow_type, const SlotD
 
 Status ParquetScanner::convert_array_to_column(ConvertFunc conv_func, size_t num_elements, const arrow::Array* array,
                                                const TypeDescriptor* type_desc, const ColumnPtr& column) {
+    if (type_desc->type == TYPE_ARRAY) {
+        // only base types are supported, nested types are not supported
+        size_t depth_limit = type_desc->get_array_depth_limit();
+        auto arrow_list_check_depth_func = get_arrow_list_check_depth();
+        Status st = arrow_list_check_depth_func(array, depth_limit);
```
Comment (chaoyli) [inline be/src/exec/parquet_scanner.cpp:251]:
```suggestion
        RETURN_IF_ERROR(arrow_list_check_depth_func(array, depth_limit));
```

File: be/test/exec/test_data/parquet_data/generate_nest_array.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#
+from pyarrow import json
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+# Generate parquet file for testing
+
+output = "./nested_array.parquet"
+
+data = [
```
Comment (chaoyli) [inline be/test/exec/test_data/parquet_data/generate_nest_array.py:23]: This file is not need.

File: be/src/exec/arrow_to_starrocks_converter.cpp [hunk 4]
```diff
@@ -779,181 +779,192 @@ struct ArrowListConverter {
...
-        if (!status.ok()) {
-            return status;
+    template <typename T>
+    static arrow::Array* get_list_array_child(const arrow::Array* array, size_t array_start_idx, size_t num_elements,
+                                              size_t* child_array_start_idx = nullptr,
+                                              size_t* child_array_num_elements = nullptr) {
+        using ArrowArrayType = typename arrow::TypeTraits<T>::ArrayType;
+        using OffsetsType = typename T::offset_type;
+        if (child_array_start_idx && child_array_num_elements) {
+            auto child_array = down_cast<const ArrowArrayType*>(array);
```
Comment (satanson) [inline be/src/exec/arrow_to_starrocks_converter.cpp:806]: can nested list  be LargeList or FixedSizeList?

File: be/src/exec/arrow_to_starrocks_converter.cpp [hunk 5]
```diff
@@ -779,181 +779,192 @@ struct ArrowListConverter {
...
+                    child_array->value_offset(array_start_idx + num_elements) - *child_array_start_idx;
+            return child_array->values().get();
+        }
+        return down_cast<const ArrowArrayType*>(array)->values().get();
+    }
+
+    static arrow::Array* unnest_list_array(const arrow::Array* array) {
+        auto type_id = array->type_id();
+        if (is_list(type_id)) {
+            return get_list_array_child<arrow::ListType>(array, 0, 0);
```
Comment (satanson) [inline be/src/exec/arrow_to_starrocks_converter.cpp:818]: 2nd, 3rd paramaters are 0, looks weird.

File: be/src/exec/arrow_to_starrocks_converter.cpp (hunk 4 above)
Comment (xiaoyong-z) [reply be/src/exec/arrow_to_starrocks_converter.cpp:807]: add FixedSizeList in new pr

**Review by satanson — DISMISSED** [review]

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
@mergify rebase

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**